contracts/contracts/strategies/Generalized4626Strategy.sol

```diff
@@ -83,20 +113,9 @@ contract Generalized4626Strategy is InitializableAbstractStrategy {
 
     /**
      * @dev Internal method to respond to the addition of new asset / share tokens
-     * @param _asset Address of the asset to approve
-     * @param _pToken The pToken for the approval
      */
-    function _abstractSetPToken(address _asset, address _pToken)
-        internal
-        virtual
-        override
-    {
-        shareToken = IERC20(_pToken);
-        assetToken = IERC20(_asset);
-
-        // Safe approval
-        shareToken.safeApprove(platformAddress, type(uint256).max);
-        assetToken.safeApprove(platformAddress, type(uint256).max);
+    function _abstractSetPToken(address, address) internal virtual override {
```

Comment:
Should this revert, since it should not be used?

Reply:
Good question. It's still used as it does the token DAI approval to the sDAI vault.

The call tree is a little roundabout, but it goes
`Generalized4626Strategy.initialize` -> `InitializableAbstractStrategy._initialize` -> `InitializableAbstractStrategy._setPTokenAddress` -> `Generalized4626Strategy._abstractSetPToken`

`_setPTokenAddress` still does useful stuff like set the `assetToPToken` mapping and `assetsMapped` array.

But you are right in the public `setPTokenAddress` is no longer used. I'll override it in `Generalized4626Strategy` to revert if called.

Reply:
I've pushed the change to override `setPTokenAddress` in `Generalized4626Strategy`. This does create a different concern in I had to make  `InitializableAbstractStrategy.setPTokenAddress` virtual so it could be overridden. There is a risk someone implementing `InitializableAbstractStrategy` will override `setPTokenAddress` but forget to set the appropriate access. eg `onlyGovernor`

What do you think? Leave the override? Or change `Generalized4626Strategy.initialize` so it doesn't call `InitializableAbstractStrategy._initialize` so we can revert in `Generalized4626Strategy._setPTokenAddress`? `Generalized4626Strategy._initialize` will then need to set the `assetToPToken` mapping and `assetsMapped` array.

Reply:
I guess leave the override, but perhaps add `onlyGovernor` to it here. This way we can use an automated tool to check that every non-view action on a strategy is non-public.

Access control is something we have to check on every PR, and hopefully we can make a tool to help double check us.

Reply:
Adding the `onlyGovernor` modifier to the `Generalized4626Strategy. setPTokenAddress` is a good idea. Thanks.

FYI, the sol2uml squash diagrams will show which contract is implementing a function and any modifiers it has. For example
![Generalized4626StrategySquashed](https://github.com/OriginProtocol/origin-dollar/assets/6491112/5e34f1a2-bcad-49c6-ba10-b404e7579ff6)

Reply:
To be consistent, I've also added an override of `Generalized4626Strategy.removePToken` to revert with the `onlyGovernor` modifier